src/stdlib_experimental_io.f90

```diff
@@ -239,8 +240,8 @@ integer function number_of_columns(s)
  do
     read(s, '(a)', advance='no', iostat=ios) c
     if (ios /= 0) exit
-    if (lastwhite .and. .not. whitechar(c)) number_of_columns = number_of_columns + 1
-    lastwhite = whitechar(c)
+    if (lastwhite .and. .not. is_blank(c)) number_of_columns = number_of_columns + 1
+    lastwhite = is_blank(c)
```

Comment:
Would it be perhaps more meaningful to rename `lastwhite` as `lastblank`?

Reply:
> Would it be perhaps more meaningful to rename `lastwhite` as `lastblank`?

@ivan-pi Thanks for the suggestion. I just pushed your proposition.